File: apps/website/mfe/src/app/pages/instant-quote/models/material.model.ts
import { IResponseModel } from './response.model';

export interface IMaterialModel {
  name: string;
  percentagePrice: number;
  pros: Array<string>;
  cons: Array<string>;
  isSelected?: boolean;
}

export interface MaterialResponse extends IResponseModel {
  data: Array<MaterialModel>;
}

export interface MaterialModel {
  name: string;
  percentagePrice: number;
  pros: Array<string>;
  cons: Array<string>;
}

File: apps/api/Models/MaterialModel.cs
using Google.Cloud.Firestore;

namespace Material.Model;


[FirestoreData]
public class MaterialModel
{
    [FirestoreProperty("id")]
    public string? ID { get; set; }
    [FirestoreProperty("name")]
    public string? Name { get; set; }

    [FirestoreProperty("percentagePrice")]
    public long? PercentagePrice { get; set; }

    [FirestoreProperty("pros")]
    public List<object>? Pros { get; set; }

    [FirestoreProperty("cons")]
    public List<object>? Cons { get; set; }
}
